src/qagame/qagame.c

```diff
@@ -194,7 +110,45 @@ void G_Say_Hook(new_gentity_t *ent, gentity_t *target, int mode, const char *cha
     // good idea to instead call the ignore checking function with the right
     // client number (ie. `ent - g_entities` or `ent->client - level.clients`).
     clientNum = ent->client->ps.clientNum;
-    ent->client->ps.clientNum = ent - (new_gentity_t *)g_entities;
+    ent->client->ps.clientNum = ent - g_entities;
     G_Say(ent, target, mode, chatText);
     ent->client->ps.clientNum = clientNum;
-}
\ No newline at end of file
+}
+
+void target_laser_think_Hook(gentity_t *self)
+{
+    vec3_t end;
+    trace_t tr;
+    vec3_t point;
+
+    // if pointed at another entity, set movedir to point at it
+    if (self->enemy)
+    {
+        VectorMA(self->enemy->s.origin, 0.5, self->enemy->r.mins, point);
+        VectorMA(point, 0.5, self->enemy->r.maxs, point);
+        VectorSubtract(point, self->s.origin, self->movedir);
+        VectorNormalize(self->movedir);
+    }
+
+    // fire forward and see what we hit
+    VectorMA(self->s.origin, 2048, self->movedir, end);
+
+    trap_Trace(&tr, self->s.origin, NULL, NULL, end, self->s.number, CONTENTS_SOLID | CONTENTS_BODY | CONTENTS_CORPSE);
+
+    // Quake III Arena and DeFRaG incorrectly check `tr.entityNum != 0`.
+    // Causing client 0 to not hurt from laser. Should actually be what is below.
+    if (tr.entityNum != ENTITYNUM_NONE)
+    {
+        // hurt it if we can
+        // breadsticks changed attacker from self->activator to self
+        // I have no idea why he did that, and neither does he.
+        // Shooters use self too, so whatever, breadsticks is never wrong.

```

Comment:
```suggestion
        // Shooters use self too, so whatever, breadsticks is never wrong [citation needed].

```

Reply:
Oh cool, didn't know you could make suggestions or that I could commit them right from this page.